source/particle/interpolator/cell_average.cc

```diff
@@ -74,8 +75,10 @@ namespace aspect
                 ghost_particles.equal_range(cell_index);
 
         const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
-        const unsigned int n_properties = particles.begin()->second.get_properties().size();
-        std::vector<double> cell_properties (n_properties,0.0);
+        const unsigned int n_particle_properties = particles.begin()->second.get_properties().size();
+        const unsigned int n_selected_properties = selected_properties.n_selected_components(n_particle_properties);
+
+        std::vector<double> cell_properties (n_selected_properties,0.0);
```

Comment:
Yes I can do that. I thought of both options and went this way because it seemed more reasonable in the beginning (you only get as many components as you asked for), but while implementing I realized it means translating indices quite often. I will update and reupload.